lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -256,63 +256,59 @@ def add_level_two_and_three_data(xml, amount, data)
         xml.invoiceReferenceNumber data[:invoice_reference_number] if data.include?(:invoice_reference_number)
         xml.customerReference data[:customer_reference] if data.include?(:customer_reference)
         xml.cardAcceptorTaxId data[:card_acceptor_tax_id] if data.include?(:card_acceptor_tax_id)
-
         {
-          sales_tax: 'salesTax',
+          tax_amount: 'salesTax',
           discount_amount: 'discountAmount',
           shipping_amount: 'shippingAmount',
           duty_amount: 'dutyAmount'
         }.each do |key, tag|
           next unless data.include?(key)
 
           xml.tag! tag do
-            data_amount = data[key].symbolize_keys
-            add_amount(xml, data_amount[:amount].to_i, data_amount)
+            add_amount(xml, data[key].to_i, data)
           end
         end
 
-        xml.discountName data[:discount_name] if data.include?(:discount_name)
-        xml.discountCode data[:discount_code] if data.include?(:discount_code)
+        # xml.discountName data[:discount_name] if data.include?(:discount_name)
+        # xml.discountCode data[:discount_code] if data.include?(:discount_code)
 
-        add_date_element(xml, 'shippingDate', data[:shipping_date]) if data.include?(:shipping_date)
+        # add_date_element(xml, 'shippingDate', data[:shipping_date]) if data.include?(:shipping_date)
 
-        if data.include?(:shipping_courier)
-          xml.shippingCourier(
-            data[:shipping_courier][:priority],
-            data[:shipping_courier][:tracking_number],
-            data[:shipping_courier][:name]
-          )
-        end
+        # if data.include?(:shipping_courier)
+        #   xml.shippingCourier(
+        #     data[:shipping_courier][:priority],
+        #     data[:shipping_courier][:tracking_number],
+        #     data[:shipping_courier][:name]
+        #   )
+        # end
 
         add_optional_data_level_two_and_three(xml, data)
 
-        if data.include?(:item) && data[:item].kind_of?(Array)
-          data[:item].each { |item| add_items_into_level_three_data(xml, item.symbolize_keys) }
-        elsif data.include?(:item)
-          add_items_into_level_three_data(xml, data[:item].symbolize_keys)
-        end
+        data[:item].each { |item| add_items_into_level_three_data(xml, item.symbolize_keys, data) } if data.include?(:item)
```

Comment:
This implies that where `data.item` was once an optional hash OR array, it is now a required array. Is it so?
Also, does this possibly mean the key can be changed to `:items`? I've been this way before and have sad memories of the time I spent staring at `data[:item].kind_of?(Array)` like _derrrr it's just item, not items_

Reply:
Yeah I don't think it makes sense to treat line_items as a hash if there's only one item. I can update it to expect `line_items` to be more clear